cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OAuth2TokenAuthenticator.java

```diff
@@ -39,10 +39,15 @@ public synchronized Request authenticate(Route route, Response response) throws
         }
         LOGGER.log(Level.FINE, "OAuth2 Token refreshed");
         // Retry the request with the new token.
+        return newRequestWithAccessTokenAsHeader(response, updatedToken);
+    }
+
+    //package scoped for testing
+    Request newRequestWithAccessTokenAsHeader(Response response, OAuth2Token oauth2Token) {
         return response.request()
             .newBuilder()
             .removeHeader(AUTHORIZATION_HEADER)
-            .addHeader(AUTHORIZATION_HEADER, "Bearer " + updatedToken.getAccessToken())
+            .addHeader(AUTHORIZATION_HEADER, oauth2Token.getTokenType() + " " + oauth2Token.getAccessToken())
```

Comment:
Assuming we can always put the token into the header? This is valid, just asking

Reply:
If someone gives you a system where the Token *DOESN'T* go in a header... run.

Unless you're thinking of something else?

Reply:
RMA has worked with a USBR service that had it as a parameter, so I asked Bryson to make sure his design allows for that but to not implement it currently as there are no CWMS web services that require that.

Reply:
I would say that if we need to provide an Authenticator that adds it as a parameter, then we can implement an Authenticator that does that, when needed.

Reply:
> RMA has worked with a USBR service that had it as a parameter, so I asked Bryson to make sure his design allows for that but to not implement it currently as there are no CWMS web services that require that.

... And *THAT* is how credentials leak. We won't be putting credentials in query parameters... ever. (for our services.)